### PR TITLE
Helps the world initialization be slightly fast.

### PR DESCRIPTION
**Pull request by eswordthecat**

### CODE REVIEW

File: code/game/atoms.dm [hunk 1]
```diff
@@ -649,4 +649,20 @@ its easier to just keep the beam vertical.
...
 
 /atom/proc/mop_act(obj/item/weapon/mop/M, mob/user)
-	return 0
\ No newline at end of file
+	return 0
+
+/atom/proc/get_areaa()
+
+/area/get_areaa()
+	return src
```
Comment (clusterfack) [inline code/game/atoms.dm]: Don't we intentionally avoid doing it this way due to lighting subareas messing up areas? Or is my understanding of area code missing something

Comment (ComicIronic) [inline code/game/atoms.dm]: How is this not the same functionality as get_area anyways?

Comment (eswordthecat) [inline code/game/atoms.dm]: We OOP now

Comment (ComicIronic) [inline code/game/atoms.dm]: So why leave get_area() in?
No duplicate procs.

Comment (eswordthecat) [inline code/game/atoms.dm]: I'm lazy i'll just do it on another pr

File: code/game/machinery/doors/firedoor.dm [hunk 2]
```diff
@@ -72,31 +72,36 @@
...
 
-	for(var/direction in cardinal)
-		A = get_area(get_step(src,direction))
-		if(istype(A) && !(A in areas_added))
-			A.all_doors.Add(src)
-			areas_added += A
+		for (var/direction in cardinal)
+			T = get_step(src, direction)
 
+			if (T)
```
Comment (ComicIronic) [inline code/game/machinery/doors/firedoor.dm]: You have to add the air_pass check to this as well, firelocks are adding incorrectly.

File: code/game/atoms.dm (hunk 1 above)
Comment (ComicIronic) [inline code/game/atoms.dm]: This is how the codesmell begins.

And if you don't want to replace all those instances, just replace get_area() with this.

Comment (eswordthecat) [inline code/game/atoms.dm]: You do realize I can just do it post haste. You see it's like i'm pseudo testing this with firelocks at live serb, if this is a success then I just replace it.

File: code/game/machinery/doors/firedoor.dm (hunk 2 above)
Comment (eswordthecat) [inline code/game/machinery/doors/firedoor.dm]: Someone will do it, so not me.

**Comment by eswordthecat** [comment]
merge when

**Comment by ComicIronic** [comment]
>duplicate procs

Literally never

**Pull request closed without merging**